### ISSUE TITLE
Segmentation fault on Ubuntu

### ISSUE THREAD

**Issue by myrrc** [philj56/tofi#85]
Log:

```
[    real,      cpu,   maxRSS]
[0.000000, 0.007320,  5468 KB][DEBUG]: This is tofi.
[0.000039, 0.007359,  5468 KB][DEBUG]: Config done
[0.000048, 0.007367,  5468 KB][DEBUG]: Connecting to Wayland display.
[0.000070, 0.007389,  5468 KB][DEBUG]: Creating xkb context.
[0.000086, 0.007405,  5468 KB][DEBUG]: First roundtrip start.
[0.000315, 0.007434,  5468 KB][DEBUG]:     Bound to shm 1.
[0.000324, 0.007443,  5468 KB][DEBUG]:     Bound to compositor 4.
[0.000328, 0.007447,  5468 KB][DEBUG]:     Bound to data device manager  6.
[WARNING]: Using an outdated compositor, screen anchoring may not work.
[0.000333, 0.007452,  5468 KB][DEBUG]:     Bound to zwlr_layer_shell_v1 12.
[0.000338, 0.007457,  5468 KB][DEBUG]:     Bound to seat 28.
[WARNING]: Using an outdated compositor, output selection will not work.
[0.000344, 0.007463,  5468 KB][DEBUG]:     Bound to output 30.
[0.000346, 0.007465,  5468 KB][DEBUG]: First roundtrip done.
[0.000348, 0.007467,  5468 KB][DEBUG]: Second roundtrip start.
[0.000377, 0.007478,  5468 KB][DEBUG]:     Got keyboard from seat.
[0.000380, 0.007481,  5468 KB][DEBUG]:     Got pointer from seat.
[0.000383, 0.007484,  5468 KB][DEBUG]:     Output configuration done.
[0.000386, 0.007487,  5468 KB][DEBUG]: Second roundtrip done.
[0.000387, 0.007489,  5468 KB][DEBUG]: Determining output.
[0.000391, 0.007492,  5468 KB][DEBUG]:     First dummy roundtrip start.
[0.000467, 0.007503,  5468 KB][DEBUG]:         Configuring keyboard.
[0.001318, 0.008355,  7948 KB][DEBUG]:         Keyboard configured.
[0.001325, 0.008361,  7948 KB][DEBUG]:         Key repeat every 40 ms after 600 ms.
[0.001330, 0.008366,  7948 KB][DEBUG]:     First dummy roundtrip done.
[0.001332, 0.008368,  7948 KB][DEBUG]:     Initialising dummy surface.
[0.001344, 0.008380,  7948 KB][DEBUG]:         Created shm file with size 0 KiB.
[0.001348, 0.008384,  7948 KB][DEBUG]:     Dummy surface initialised.
[0.001350, 0.008386,  7948 KB][DEBUG]:     Second dummy roundtrip start.
[0.001546, 0.008402,  7948 KB][DEBUG]:     Second dummy roundtrip done.
[0.001559, 0.008415,  7948 KB][DEBUG]: Selected output .
[0.001562, 0.008418,  7948 KB][DEBUG]: Generating command list.
[0.001564, 0.008419,  7948 KB][DEBUG]:     Retrieving PATH.
[0.001566, 0.008422,  7948 KB][DEBUG]:     Retrieving cache location.
[0.001583, 0.008439,  7948 KB][DEBUG]:     Cache up to date, loading.
[0.001677, 0.008532,  7948 KB][DEBUG]:     Moving already known programs to the front.
[0.001696, 0.008552,  7948 KB][DEBUG]: Command list generated.
[0.001706, 0.008561,  7948 KB][DEBUG]: Creating main window surface.
zsh: segmentation fault (core dumped)  tofi-run
```

Installation:

```sh
sudo apt install libfreetype-dev libcairo2-dev libpango1.0-dev libwayland-dev wayland-protocols libxkbcommon-dev scdoc
 CC=clang-15 meson build --prefix ~/.local; ninja -C build install  
tofi-run
```

lldb:

```
* thread #1, name = 'tofi-run', stop reason = signal SIGSEGV: invalid address (fault address: 0x0)
    frame #0: 0x00007ffff77934aa libwayland-client.so.0`wl_proxy_marshal_constructor + 138
libwayland-client.so.0`wl_proxy_marshal_constructor:
->  0x7ffff77934aa <+138>: movq   (%rbp), %rdx
    0x7ffff77934ae <+142>: leaq   0x20(%rsp), %r14
    0x7ffff77934b3 <+147>: leaq   0x8(%rsp), %rcx
    0x7ffff77934b8 <+152>: leaq   0x1b0(%rsp), %rax
```

System: Ubuntu 20.04
Sway: 1.4

**Comment by myrrc** [comment]
Meson build log

```
The Meson build system
Version: 1.0.1
Source dir: /home/myrrc/tofi
Build dir: /home/myrrc/tofi/build
Build type: native build
Project name: tofi
Project version: 0.8.1
C compiler for the host machine: clang-15 (clang 15.0.7 "Ubuntu clang version 15.0.7")
C linker for the host machine: clang-15 ld.bfd 2.34
Host machine cpu family: x86_64
Host machine cpu: x86_64
Library rt found: YES
Library m found: YES
Checking for function "fts_read" : YES
Library fts found: NO
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Run-time dependency freetype2 found: YES 23.1.17
Run-time dependency harfbuzz found: YES 2.6.4
Run-time dependency cairo found: YES 1.16.0
Run-time dependency pangocairo found: YES 1.44.7
Run-time dependency wayland-client found: YES 1.18.0
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Build-time dependency wayland-protocols found: YES 1.20
Build-time dependency wayland-scanner found: YES 1.18.0
Run-time dependency xkbcommon found: YES 0.10.0
Run-time dependency glib-2.0 found: YES 2.64.6
Run-time dependency gio-unix-2.0 found: YES 2.64.6
Program /usr/bin/wayland-scanner found: YES (/usr/bin/wayland-scanner)
Program scdoc found: YES (/usr/bin/scdoc)
Program sed found: YES (/usr/bin/sed)
Program sh found: YES (/usr/bin/sh)
Build targets in project: 14

tofi 0.8.1

  User defined options
    prefix: /home/myrrc/.local

Found ninja-1.10.0 at /usr/bin/ninja
```

**Comment by myrrc** [comment]
According to the log

```
[WARNING]: Using an outdated compositor, screen anchoring may not work.
[0.000295, 0.007042,  5484 KB][DEBUG]:     Bound to zwlr_layer_shell_v1 12.
[0.000301, 0.007048,  5484 KB][DEBUG]:     Bound to seat 28.
```

branch at https://github.com/philj56/tofi/blob/e6b9b576b96e87a29047a3951ab2a6838e1df781/src/main.c#L692

is taken, `wp_vieporter` is not set, but then 
https://github.com/philj56/tofi/blob/e6b9b576b96e87a29047a3951ab2a6838e1df781/src/main.c#L1584

is invoked with null `wp_viewporter`

**Comment by myrrc** [comment]
git bisect `rm -fr build; CC=clang-15 meson build -Dbuildtype=debug --prefix ~/.local; ninja -C build install; tofi-run` showed that the bad commit is
[8287](https://github.com/philj56/tofi/commit/828744849d0b41d0b65db4f0bcb9be7e8698fdd0)

**Comment by philj56** [comment]
Hey, thanks for the detailed report! I had assumed viewporter would be available on Ubuntu 20.04, as the protocol was made stable in 2016, but apparently Sway didn't implement it until 2020. Should be a quick fix.

**Comment by philj56** [comment]
Alright, should be fixed in c6f539e065ca1a6f9a9048ce211132891afbbe6b.
